	# Fix bug 33696 by allowing usernames longer than 8 chars,
	# thanks to Yuval Kogman for the patch
--- at-3.1.8/atd.c
+++ at-3.1.8/atd.c
@@ -196,7 +196,7 @@
  */
     pid_t pid;
     int fd_out, fd_in;
-    char mailbuf[9], jobbuf[9];
+    char mailbuf[256], jobbuf[9];
     char *mailname = NULL;
     char *newname;
     FILE *stream;
@@ -299,7 +299,7 @@
      * NFS and works with local file systems.  It's not clear where
      * the bug is located.  -Joey
      */
-    if (fscanf(stream, "#!/bin/sh\n# atrun uid=%d gid=%d\n# mail %8s %d",
+    if (fscanf(stream, "#!/bin/sh\n# atrun uid=%d gid=%d\n# mail %255s %d",
 	       &nuid, &ngid, mailbuf, &send_mail) != 4)
 	pabort("File %.500s is in wrong format - aborting",
 	       filename);
